docs/developer.md

```diff
@@ -0,0 +1,203 @@
+# Developer Docs
+
+In order to contribute to this project, make sure you have go 1.18 installed
+on your local machine.
+
+## Dependencies
+
+The build system assumes the following binaries are in the `PATH`:
+
+```
+make git go npm kind podman (or docker)
+```
+Once these tools are installed, run `make tools` to install all required
+project dependencies to ``tmp/bin``
+```
+make tools
+```
+## Development Environment Setup
+
+To setup your development, it is recommended to run the helper script
+`hack/kind/setup.sh`. This script does the following
+    * sets up a Kind cluster
+    * installs OLM
+    * sets up a local registry to push locally built operator and bundle images
+
+```sh
+./hack/kind/setup.sh
+```
+
+Once done, the cluster can be deleted by running
+
+```
+kind delete cluster --name obs-operator
+```
+
+## Running End to End tests
+
+Running E2E locally against the kind cluster that was setup following the
+instructions above is as easy as running `./test/run-e2e.sh`.
+
+```sh
+ ./test/run-e2e.sh
+```
+**NOTE:** `./test/run-e2e.sh --help` shows options that are useful when
+rerunning tests.
+
+
+# Running Operator In Kind Cluster
+
+
+Run `./hack/kind/setup.sh` to create a Kind cluster named `obs-operator` and
+to set up all the required dependencies (OLM, infra-node, registry) in the
+cluster.
+
+To cleanup the cluster or to retry the setup script after a failed attempt,
+run `kind delete cluster --name obs-operator`
+
+## Deploy ObO using `operator-sdk`
+
+Observability Operator relies heavily on (forked) Promethues Operator to do
```

Comment:
Not entirely sure how this paragraph of information relates to the title `Deploy ObO using  operator-sdk` maybe I'm missing something. I think my confusion comes from the many mentions of `operator` in the text, and sometimes it's not clear which operator we are referring to

Reply:
To deploy some thing to k8s, why use `operator-sdk` is a question the reader may have. 
The usual approach is to just `kubectl create -f <dir>` - right? So I thought this warrants an explanation. 

But I agree, we should rephrase it if it is confusing.